node/operations.js

```diff
@@ -85,9 +87,25 @@ OperationTombstone.prototype.extendLogInfo = function extendLogInfo(info) {
     return info;
 };
 
+OperationTombstone.prototype.destroy = function destroy(now) {
+    var self = this;
+
+    self.destroyed = true;
+
+    self.onTimeout(now);
+};
+
 OperationTombstone.prototype.onTimeout = function onTimeout(now) {
     var self = this;
 
+    if (!self.destroyed && now < self.timeout + self.time) {
```

Comment:
Early time-outs are a thing in the destroy() path. Which is why I have a destroyed gaurd.

This explains why the client and relay have different number of tombstones in the benchmark. 

I don't think we have a heap bug.